packages/chat-headless-react/tests/headlessProvider.test.tsx

```diff
@@ -0,0 +1,63 @@
+import { render } from "@testing-library/react";
+import {
+  ChatHeadlessProvider,
+  ConversationState,
+  HeadlessConfig,
+} from "../src";
+import { renderToString } from "react-dom/server";
+import { useChatState } from "../src/useChatState";
+
+it("only fetches session storage on client-side render", async () => {
+  const win = window;
+  Object.defineProperty(win, "sessionStorage", {
+    value: {
+      ...win.sessionStorage,
+      getItem: (_: string): string => {
+        return JSON.stringify({
+          messages: [{ text: "foobar", source: "BOT" }],
+          isLoading: false,
+          canSendMessage: false,
+        } satisfies ConversationState);
+      },
+    },
+  });
+  const windowSpy = jest
+    .spyOn(window, "window", "get")
+    .mockImplementation(() => win);
+  const config: HeadlessConfig = {
+    botId: "123",
+    apiKey: "1234",
+    saveToSessionStorage: true,
+  };
+  const str = () =>
```

Comment:
nit: might be simpler to directly invoke `renderToString` in line 40:
```
container.innerHTML = renderToString(
      <ChatHeadlessProvider config={config}>
        <TestComponent />
      </ChatHeadlessProvider>
    );
```

Reply:
I think I would prefer to keep these separate, search's tests do the same paradigm